MovieTime/MovieTime/Core/Router/MTRouter.swift

```diff
@@ -0,0 +1,65 @@
+//
+//  MTRouter.swift
+//  MovieTime
+//
+//  Created by Enrique Melgarejo on 10/01/18.
+//  Copyright © 2018 Choynowski. All rights reserved.
+//
+
+import UIKit
+
+enum MTAppState {
+    case splashScreen
+    case feed
+}
+
+final class MTRouter: NSObject {
+
+    static let shared = MTRouter()
+
+    private var currentState = MTAppState.splashScreen
+
+    private func changeRootViewController(forIdentifier identifier: String, storyboardName: String) {
+
+        guard let appDelegate = UIApplication.shared.delegate,
+            let window = appDelegate.window else { return }
+
+        let storyboard = UIStoryboard.init(name: storyboardName, bundle: nil)
+        let desiredViewController = storyboard.instantiateViewController(withIdentifier: identifier)
+
+        if let snapshot: UIView = (window?.snapshotView(afterScreenUpdates: true)) {
+
+            desiredViewController.view.addSubview(snapshot)
+            window?.rootViewController = desiredViewController
+
+            UIView.animate(withDuration: 0.3, animations: {() in
```

Comment:
`() in` can probably be omitted, and if not, `_ in` is a more common notation. That's why I think it can be omitted, the underscore exists to omit names, if `()` was accepted then it probably means this closure has no parameter